### ISSUE TITLE
Issue with saving the best model state

### ISSUE THREAD

**Issue by yuanx749** [WenjieDu/PyPOTS#667]
### Issue description

Hi, thanks for this great tool!

I believe the line below should be `self.best_model_dict = deepcopy(self.model.state_dict())`. Otherwise the best model keeps getting updated, because `model.state_dict()` returns a reference to the dict.
https://github.com/WenjieDu/PyPOTS/blob/187852a5fe201da4550649362b7de309f38bfedc/pypots/base.py#L782
I encountered this issue when training a model with more epochs, I got different results even though the best model is at the same epoch.

### Your contribution

As suggested in the description.

**Comment by WenjieDu** [comment]
Hi @yuanx749, thanks for raising this issue. I agree with you. I'm going to label this issue as `bug`. Would you like to make a PR to fix this? This could be your first contribution to PyPOTS ;-)

**Comment by yuanx749** [comment]
Hi @WenjieDu , thanks for your quick response. I'll look into this next week, but since it's a small change, you can go ahead and make the fix if you'd like. :)

**Comment by WenjieDu** [comment]
No problem, Xiao. I've created PR #668 to fix it. Please have a review ;~)
